.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,32 @@
+name: "CodeQL"
+# see https://codeql.github.com/docs/
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
+
+jobs:
+  analyse:
+    name: Analyse
+    runs-on: ubuntu-latest
```

Comment:
AIUI, that "ubuntu-latest" was to indicate that the Ubuntu 20.10 and later support CodeQL.

FYI, locking the test to a specific OS version will prevent the ongoing compiler advancements on static analysis being available. This could easily result in the ```master``` code testing fine but then failing to build on slightly newer compilers.

Reply:
> locking the test to a specific OS version will prevent the ongoing compiler advancements on static analysis being available

I disagree. We can always change the OS version. The question is who is in control of that change: We or GitHub. I believe it should be us. I bet we will know about new OS availability soon enough and can plan an upgrade on our terms at that time.